src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private MyNode<T> head;
+    private MyNode<T> tail;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            addFirst(value);
+            return;
+        }
+        addLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        if (index == 0) {
+            head.prev = new MyNode<>(null, value, head);;
+            head = head.prev;
+        } else {
+            MyNode<T> current = getNode(index);
+            MyNode<T> newNode = new MyNode<>(current.prev, value, current);
+            current.prev.next = current.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            addLast(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        MyNode<T> oldNode = getNode(index);
+        T oldItem = oldNode.item;
+        oldNode.item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        MyNode<T> node = getNode(index);
+        unlink(node);
+        return node.item;
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (MyLinkedList.MyNode<T> node = head; node != null; node = node.next) {
```

Comment:
such using of for-loop is not standardized and non-obvious to read. 

I would suggest using while-loop for such cases.